lib/builder/components/ShareDashboard.js

```diff
@@ -0,0 +1,102 @@
+import React, { useEffect } from "react";
+import { connect } from "react-redux";
+import { withRouter } from "react-router-dom";
+import {
+  toogleDashboardsMenu,
+  setAccessKey,
+  clearAccessKey,
+} from "../../actions/rootActions";
+import client from "../../../config/analysis";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+
+const ShareDashboards = props => {
+  useEffect(() => {
+    const getKeyFromAPI = async () => {
+      const keyName = `public-dashboard:${props.dashboardTitle}`;
+      let key = await
+        client
+          .get({
+            url: client.url('projectId', `keys?name=${keyName}`),
+            api_key: client.masterKey(),
+          });
+        
+        if (!key.length) {
+          key = await
+            client
+              .post({
+                url: client.url('projectId', 'keys'),
+                api_key: client.masterKey(),
+                params: {
+                  name: keyName,
+                  is_active: true,
+                  permitted: [ 'queries', 'saved_queries' ],
+                }
+              })
+          return key.key;
+        }
+        return key[0].key;
+    }
+
+    getKeyFromAPI().then(data => props.setAccessKey(data));
+
+  }, []);
+
+  const shareUrl = `https://keen.github.io/dashboard-builder/#/viewer/?dashboardId=${props.id}&accessKey=${props.accessKey}`;
```

Comment:
it would be beneficial to read the core part of these url (domain) from the config. The app should detect if it's hosted on gh-pages and use this one, or if it's hosted on keen-web (then use dashboards.keen.io url). Related to: https://github.com/keenlabs/Keen-Web/issues/2453